frame/contracts/src/exec.rs

```diff
@@ -1092,6 +1096,19 @@ where
 		secp256k1_ecdsa_recover_compressed(&signature, &message_hash).map_err(|_| ())
 	}
 
+	fn set_code_hash(&mut self, hash: CodeHash<Self::T>) -> Result<(), DispatchError> {
+		increment_refcount::<Self::T>(hash)?;
+		let top_frame = self.top_frame_mut();
+		let prev_hash = top_frame.contract_info().code_hash.clone();
+		top_frame.contract_info().code_hash = hash;
+		decrement_refcount::<Self::T>(prev_hash)?;
```

Comment:
It does not matter for correctness because we roll back on error anyways but we should do it never the less as erroring out as early as possible is still the right thing to do. As a matter of fact I did already suggest that but it was glossed over.

Reply:
@athei, sorry I forgot to move up decrement_refcount. Done